lib/componentmanager.js

```diff
@@ -302,6 +302,19 @@
         this._metaDefaultComponents.push(component);
         this._allComponents[componentId] = component;
     };
+
+    ComponentManager.prototype.addDocumentComponent = function (component) {
+        var componentId = this.getComponentId();
+        component.id = componentId;
+        component.assetPath = _getAssetPath(component);
+
+        this._documentComponent = component;
```

Comment:
need to check if this is already set? Also should this be an array? (That settings were an array before?)